internal/service/service_windows.go

```diff
@@ -124,11 +126,32 @@ func (ws *windowsService) Run(onStart, onStop func() error) error {
 
 	ws.logger = elog
 
+	ok, err := svc.IsWindowsService()
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return ws.runNotSerVice(onStart, onStop)
+	}
```

Comment:
I am not sure what's happening here. Could you add a comment explaining why we need to do this?

Reply:
"service mode": mg is executed by windows as a service.
"normal mode": mg is executed by user. e.g. mg daemon run.
`svc.Run` can only be called in "service mode".